docs/operation/disk-management.md

```diff
@@ -0,0 +1,185 @@
+
+**Note**: The disk monitor with forced retention cleanup is only available for **standalone** and **data** servers. **Liaison** servers use the `*-max-disk-usage-percent` flags to throttle writes when the write queue fills up, allowing the sync process to catch up with incoming data.
+
+## Configuration
+
+### Data & Standalone Servers
+
+The following flags are used to configure forced retention cleanup for data and standalone servers:
+
+#### Measure Service
+
+- `--measure-retention-high-watermark float`: Disk usage percentage that triggers forced retention cleanup (0-100, default: 95.0).
+- `--measure-retention-low-watermark float`: Disk usage percentage where forced retention cleanup stops (0-100, default: 85.0).
+- `--measure-retention-check-interval duration`: Interval for checking disk usage (default: 5m).
+- `--measure-retention-cooldown duration`: Cooldown period between forced segment deletions (default: 30s).
+
+#### Stream Service
+
+- `--stream-retention-high-watermark float`: Disk usage percentage that triggers forced retention cleanup (0-100, default: 95.0).
+- `--stream-retention-low-watermark float`: Disk usage percentage where forced retention cleanup stops (0-100, default: 85.0).
+- `--stream-retention-check-interval duration`: Interval for checking disk usage (default: 5m).
+- `--stream-retention-cooldown duration`: Cooldown period between forced segment deletions (default: 30s).
+
+#### Trace Service
+
+- `--trace-retention-high-watermark float`: Disk usage percentage that triggers forced retention cleanup (0-100, default: 95.0).
+- `--trace-retention-low-watermark float`: Disk usage percentage where forced retention cleanup stops (0-100, default: 85.0).
+- `--trace-retention-check-interval duration`: Interval for checking disk usage (default: 5m).
+- `--trace-retention-cooldown duration`: Cooldown period between forced segment deletions (default: 30s).
+
+### Liaison Servers
+
+Liaison servers use the disk usage flags to manage their write queue and prevent disk space exhaustion:
+
+- `--measure-max-disk-usage-percent int`: The maximum disk usage percentage allowed for measure data (0-100, default: 95).
+- `--stream-max-disk-usage-percent int`: The maximum disk usage percentage allowed for stream data (0-100, default: 95).
+- `--trace-max-disk-usage-percent int`: The maximum disk usage percentage allowed for trace data (0-100, default: 95).
+
+**Write Queue Mechanism**: Liaison servers maintain a write queue that buffers incoming data before syncing it to data servers. When the queue fills up and disk usage exceeds the configured threshold, the liaison server throttles incoming writes to allow the sync process to catch up and free up disk space.
+
+### Property Service
+
+- `--property-max-disk-usage-percent int`: The maximum disk usage percentage allowed (0-100, default: 95).
```

Comment:
What is the policy of this? Reject write and the client received an error? Please be clear.

Reply:
Write throttle like other models. I will clarify this.